astropy/coordinates/representation.py

```diff
@@ -581,7 +596,7 @@ def _re_represent_differentials(self, new_rep, differential_class):
         if differential_class is None:
             return dict()
 
-        if not self.differentials:
+        if not self.differentials and differential_class:
```

Comment:
Only fail if an empty dict is passed and there are no differentials - this allows for slightly cleaner code elsewhere.